CRM/Event/Form/Task/Batch.php

```diff
@@ -417,7 +417,11 @@ public static function updateContributionStatus($params) {
     //complete the contribution.
     // @todo use the api - ie civicrm_api3('Contribution', 'completetransaction', $input);
     // as this method is not preferred / supported.
-    CRM_Contribute_BAO_Contribution::completeOrder($input, $ids, $objects);
+    CRM_Contribute_BAO_Contribution::completeOrder($input, [
+      'related_contact' => $ids['related_contact'] ?? NULL,
+      'participant' => !empty($objects['participant']) ? $objects['participant']->id : NULL,
+      'contributionRecur' => NULL,
```

Comment:
We actually know recur is null here as we don't support online recurring event payments - which is when this is called. I actually want to stop calling completeOrder here but for now this is OK